### PR TITLE
KAFKA-12393: Document multi-tenancy considerations

### PR DESCRIPTION
**Pull request by miguno**
See https://issues.apache.org/jira/browse/KAFKA-12393

### CODE REVIEW

**Comment by miguno** [comment]
cc to committer @rajinisivaram as the SME on this subject, and who has the most context

**Review by bbejeck — APPROVED** [review]
LGTM and verified locally, but I'll wait on merging for a review from @rajinisivaram 

Also, @miguno, can you create an identical PR to go against docs in AK trunk?

**Comment by miguno** [comment]
@bbejeck wrote in https://github.com/apache/kafka-site/pull/334#pullrequestreview-600990037:
> Also, @miguno, can you create an identical PR to go against docs in AK trunk?

Yes, I will do this once the content review of this PR is completed.

File: 27/ops.html
Comment (dajac) [inline 27/ops.html]: I am not sure to get your point here. While I do agree that disabling the auto topic creation is a good thing, users/apps can still create topics with the admin client so it does not really help to enforce a topic naming structure. In both cases, the topics would have to respect the ACLs in place and the "namespace" if defined.

File: 27/ops.html
Comment (dajac) [inline 27/ops.html]: `#brokerconfigs_controller.quota.window.num">` is not the correct config to highlight. `controller_mutations_rate` is more appropriate. We don't have an anchor for it though. I suggest to remove it for now.

I need to add more documentation about the controller quota. We can add a link to it here when it is done.

**Review by dajac — COMMENTED** [review]
@miguno Thanks for the doc. This is great overall. I just left few small comments.

File: 27/ops.html
Comment (rajinisivaram) [inline 27/ops.html]: Don't think we would refer to request rate or bandwidth quotas as `client-side`. They apply to clients, but are protecting broker resources and are enforced by brokers.

File: 27/ops.html
Comment (rajinisivaram) [reply 27/ops.html]: Yes, the text sounds like you need to disable auto topic creation to enforce topic ACLs which is not the case, we should rewrite that part.

File: 27/ops.html
Comment (rajinisivaram) [inline 27/ops.html]: Should we also talk about `policies` like CreateTopicPolicy/AlterConfigPolicy that also support additional restrictions?

**Review by rajinisivaram — COMMENTED** [review]
@miguno Thanks for the PR, looks good. Left a few minor comments.

File: 27/ops.html
Comment (miguno) [reply 27/ops.html]: Ack and updated

File: 27/ops.html
Comment (miguno) [reply 27/ops.html]: Yeah, I can add a note. But it's unfortunate that there's essentially zero coverage in the AK docs on how to use these.

File: 27/ops.html
Comment (miguno) [reply 27/ops.html]: Ack and updated.

**Comment by miguno** [comment]
PR updated with reviewer feedback.

**Review by dajac — APPROVED** [review]
LGTM, thanks!

**Review by rajinisivaram — APPROVED** [review]
@miguno Thanks for the updates, LGTM

**Comment by miguno** [comment]
Thanks all for reviewing, much appreciated!

**Pull request merged**
